### ISSUE TITLE
Making `org.kotools.types.text.NotBlankString` type serializable

### ISSUE THREAD

**Issue by LVMVRQUXL** [kotools/types#376]
## Description

Like the `kotools.types.text.NotBlankString` type, we would like to make the `org.kotools.types.text.NotBlankString` type serializable as `String`.

Here's an example of Kotlin code that encodes and decodes this type using the [JavaScript Object Notation (JSON) format from kotlinx.serialization](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json/kotlinx.serialization.json/-json):

```kotlin
val text: NotBlankString = NotBlankString.create("Kotools Types")
val encoded: String = Json.encodeToString(text)
println(encoded) // "Kotools Types"
val decoded: NotBlankString = Json.decodeFromString(encoded)
println(decoded == text) // true
```

## Dependencies

This issue is blocked by the following ones:

- [x] #372 

## Checklist

- [ ] Make the type serializable, [test] its serialization and deserialization processes, update the public [API binaries] and the documentation of the type.

[api binaries]: https://github.com/kotools/types/blob/main/CONTRIBUTING.md#checking-the-api-binaries
[test]: https://github.com/kotools/types/blob/main/CONTRIBUTING.md#running-tests

**Comment by LVMVRQUXL** [comment]
This issue will be implemented by the issue #372.

**Comment by LVMVRQUXL** [comment]
This issue will be implemented by the issue #372.
